### PR TITLE
update to modifield binary mera optimization

### PR DESCRIPTION
**Pull request by mganahl**
Removed some minor bugs and added some dosctrings
cleaned up the code

### CODE REVIEW

File: experiments/MERA/modified_binary_mera_lib.py [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+
+""" 
+modified binary MERA optimization
+parts of the following code are based on code written by Glen Evenbly (c) for www.tensors.net, (v1.1) 
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import sys
+sys.path.append('../../')
```
Comment (Thenerdstation) [inline experiments/MERA/modified_binary_mera_lib.py]: Don't do this. Make experiments a module when possible.

File: experiments/MERA/modified_binary_mera_lib.py [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+import tensorflow as tf
+import copy
+import numpy as np
+import time
+import pickle
```
Comment (Thenerdstation) [inline experiments/MERA/modified_binary_mera_lib.py]: Remove unused imports

File: experiments/MERA/modified_binary_mera_lib.py [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+        rhoBA[-1] = rhoBA_0        
+        for p in range(len(rhoAB)-2,-1,-1):
+            rhoAB[p], rhoBA[p] = descending_super_operator(rhoAB[p+1],rhoBA[p+1],wC[p],vC[p],uC[p],refsym)
+
+        if verbose > 0:
+            if np.mod(k,10) == 1:
+                Energies.append((ncon.ncon([rhoAB[0],hamAB[0]],[[1,2,3,4],[1,2,3,4]]) + 
+                                ncon.ncon([rhoBA[0],hamBA[0]],[[1,2,3,4],[1,2,3,4]]))/4 + bias/2)
+                stdout.write('\rIteration: %i of %i: E = %.8f, err = %.16f at D = %i with %i layers' %
+                             (int(k),int(numiter), float(Energies[-1]), float(Energies[-1] + 4/np.pi,), int(wC[-1].shape[2]), len(wC)))
```
Comment (Thenerdstation) [inline experiments/MERA/modified_binary_mera_lib.py]: Bring your line lengths to be less than 80 characters. (We will likely add this restriction to the presubmit soon.)

File: experiments/MERA/modified_binary_mera_lib.py (hunk 1 above)
Comment (mganahl) [reply experiments/MERA/modified_binary_mera_lib.py]: removed

**Comment by Thenerdstation** [comment]
> I tried that, but it's not there!

Really? Definitely works for me on master.

```python
>>> import tensornetwork
>>> tensornetwork.ncon
<function ncon at 0x7f6c78241d08>
```

**Comment by Thenerdstation** [comment]
Make sure you you run your experiment like `python3 -m experiments.<YOUR_EXPERIMENT>` from the root directory and make sure your pip install of tensornetwork is removed.

We're working with a modules instead of a single giant file now, so the workflow is going to be slightly different.

**Comment by mganahl** [comment]
I haven't run the pip installer yet. I'm working on my own fork of the repo at the moment. My PYTHONPATH is pointing to the root directory, and I'm running python 3.6 in a virtualenv. 
Do I have to set up anything else to make this work?

**Comment by Thenerdstation** [comment]
Sounds like your setup is the issue and I don't think I can fix it over GitHub comments.

Wanna setup a video chat and fix it next week?

**Comment by mganahl** [comment]
Hey, I got it working, it was a misunderstanding on my side. 'python3 modified_binary_mera.py'  is running now from the experiments/MERA folder.  Running ' python3 -m experiments.<YOUR_EXPERIMENT>' from root currently raises an `AttributeError` (has no attribute __path__). I guess I have to set up an appropriate __init__.py in experiments.MERA for that. I'll do that today or tomorrow and add it to the PR. thanks!

File: experiments/MERA/modified_binary_mera_lib.py [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+                             nsteps_steady_state=8,
+                             verbose=0,
+                             opt_u=True,
+                             opt_vw=True,
+                             numpy_update=True,
+                             opt_all_layers=False,
+                             opt_u_after=9):
+    """
+    ------------------------
+    adapted from Glen Evenbly (c) for www.tensors.net, (v1.1) - last modified 24/1/2019
```
Comment (Thenerdstation) [inline experiments/MERA/modified_binary_mera_lib.py]: Try to follow the Google style python docstring. We'll be enforcing it soon on the entire codebase.
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Comment by amilsted** [comment]
I have altered my wavefunction and tree scripts so they run via `python -m` in https://github.com/google/TensorNetwork/pull/39.

**Comment by Thenerdstation** [comment]
We can merge this once you fix the merge conflicts. We are enforcing the Google python style so sorry if there is a lot of changes. :(

**Pull request merged**